### PR TITLE
Create and expose a subscriptions database (closes #263)

### LINKED ISSUE #263: Add subscriptions database (android)

### PR DESCRIPTION
**Pull request by wilsonpage**

### CODE REVIEW

File: android/app/src/main/java/org/mozilla/magnet/database/HistorySQLTable.java [hunk 1]
```diff
@@ -150,7 +150,7 @@ private HistoryRecord toHistoryRecord(Cursor cursor) {
     private static abstract class Schema implements BaseColumns {
         public static final String TABLE_NAME = "history";
         public static final String COLUMN_NAME_URL = "url";
-        public static final String COLUMN_NAME_TIME_FIRST_SEEN = "timefirstseen";
-        public static final String COLUMN_NAME_TIME_LAST_SEEN = "timelastseen";
+        public static final String COLUMN_NAME_TIME_FIRST_SEEN = "time_first_seen";
+        public static final String COLUMN_NAME_TIME_LAST_SEEN = "time_last_seen";
```
Comment (wilsonpage) [inline android/app/src/main/java/org/mozilla/magnet/database/HistorySQLTable.java:154]: Snake case for consistency.

File: android/app/src/main/java/org/mozilla/magnet/database/SubscriptionsSQLTable.java [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+    /**
+     * Called by `DatabaseSQL`.
+     * @param db
+     */
+    public void onCreate(SQLiteDatabase db) {
+        db.execSQL("CREATE TABLE " + Schema.TABLE_NAME + "("
+                    + Schema.COLUMN_NAME_CHANNEL_NAME + " TEXT PRIMARY KEY,"
+                    + Schema.COLUMN_NAME_NOTIFICATIONS_ENABLED + " INT"
+                + ")");
```
Comment (wilsonpage) [inline android/app/src/main/java/org/mozilla/magnet/database/SubscriptionsSQLTable.java:32]: Do I need an index on this? Not a massive SQL pro.

**Comment by wilsonpage** [comment]
@arcturus r?

**Comment by wilsonpage** [comment]
I noticed that I'd left some redundant alarm related code from the last issue. So there's a patch that strips it out.

File: android/app/src/main/java/org/mozilla/magnet/database/SubscriptionsSQLTable.java [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+public class SubscriptionsSQLTable implements SubscriptionsStore, DatabaseSQL.Table {
+    private final String TAG = History.class.getName();
+    private DatabaseSQL mDB;
+
+    /**
+     * Create a new `History` object.
```
Comment (arcturus) [inline android/app/src/main/java/org/mozilla/magnet/database/SubscriptionsSQLTable.java]: ni: Create a new `Subscription` ..

File: android/app/src/main/java/org/mozilla/magnet/database/SubscriptionsSQLTable.java (hunk 2 above)
Comment (arcturus) [reply android/app/src/main/java/org/mozilla/magnet/database/SubscriptionsSQLTable.java:32]: Yes please index it :P

File: android/app/src/main/java/org/mozilla/magnet/database/SubscriptionsSQLTable.java [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+     *
+     * @param channelName
+     * @param model
+     */
+    public boolean update(String channelName, HashMap model) {
+        SQLiteDatabase db = mDB.getWritableDatabase();
+        ContentValues values = new ContentValues();
+
+        // add known keys
+        if (model.containsKey(Schema.COLUMN_NAME_NOTIFICATIONS_ENABLED)) {
```
Comment (arcturus) [inline android/app/src/main/java/org/mozilla/magnet/database/SubscriptionsSQLTable.java:97]: we should have the `else` branch to setup the column_name_notifications_enabled to 0 right?

**Review by arcturus — APPROVED** [review]
Looks good to me, just one question, does the changes related to removing the alarm need to come in this PR?

File: android/app/src/main/java/org/mozilla/magnet/database/SubscriptionsSQLTable.java (hunk 2 above)
Comment (wilsonpage) [reply android/app/src/main/java/org/mozilla/magnet/database/SubscriptionsSQLTable.java:32]: Suggests `PRIMARY KEY` is already indexed.

Comment (wilsonpage) [reply android/app/src/main/java/org/mozilla/magnet/database/SubscriptionsSQLTable.java:32]: http://stackoverflow.com/questions/3379292/is-an-index-needed-for-a-primary-key-in-sqlite

**Comment by wilsonpage** [comment]
Landing on green.

File: android/app/src/main/java/org/mozilla/magnet/database/SubscriptionsSQLTable.java (hunk 2 above)
Comment (arcturus) [reply android/app/src/main/java/org/mozilla/magnet/database/SubscriptionsSQLTable.java:32]: Great, thanks!

**Pull request merged**